include/ginkgo/core/base/types.hpp

```diff
@@ -559,6 +559,73 @@ GKO_ATTRIBUTES constexpr bool operator!=(precision_reduction x,
 #endif
 
 
+/**
+ * Instantiates a template for each non-complex value, local and global index
+ * type compiled by Ginkgo.
+ *
+ * @param _macro  A macro which expands the template instantiation
+ *                (not including the leading `template` specifier).
+ *                Should take three arguments, which are replaced by the
+ *                value, the local and the global index types.
+ */
+#if GINKGO_DPCPP_SINGLE_MODE
+#define GKO_INSTANTIATE_FOR_EACH_NON_COMPLEX_VALUE_AND_LOCAL_GLOBAL_INDEX_TYPE( \
```

Comment:
Honestly, I'm not sure how that would work. I guess it would be somewhat easy, all possible combination of local and global index type, but I wanted to not instantiate the case `local_idx_type = int64` and `global_idx_type = int32`. Do you already have something in mind for this?

Reply:
Makes sense, then nevermind :)